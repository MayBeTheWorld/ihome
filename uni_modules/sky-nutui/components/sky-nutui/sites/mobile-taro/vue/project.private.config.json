{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}