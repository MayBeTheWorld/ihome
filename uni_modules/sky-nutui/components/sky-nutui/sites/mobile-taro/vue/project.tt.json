{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
