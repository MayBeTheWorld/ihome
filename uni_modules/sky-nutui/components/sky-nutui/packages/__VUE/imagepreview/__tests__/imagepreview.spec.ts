import { mount } from '@vue/test-utils';
import ImagePreview from '../index.vue';
import { nextTick, reactive, toRefs } from 'vue';
// import { trigger } from '../../utils/unit';

const images = [
  {
    src: '//m.360buyimg.com/mobilecms/s750x366_jfs/t1/18629/34/3378/144318/5c263f64Ef0e2bff0/0d650e0aa2e852ee.jpg'
  },
  {
    src: '//m.360buyimg.com/mobilecms/s750x366_jfs/t1/26597/30/4870/174583/5c35c5d2Ed55eedc6/50e27870c25e7a82.png'
  },
  {
    src: '//m.360buyimg.com/mobilecms/s750x366_jfs/t1/9542/17/12873/201687/5c3c4362Ea9eb757d/60026b40a9d60d85.jpg'
  },
  {
    src: '//m.360buyimg.com/mobilecms/s750x366_jfs/t1/30042/36/427/82951/5c3bfdabE3faf2f66/9adca782661c988c.jpg'
  }
];

const videos = [
  {
    source: {
      src: 'https://storage.jd.com/about/big-final.mp4?Expires=3730193075&AccessKey=3LoYX1dQWa6ZXzQl&Signature=ViMFjz%2BOkBxS%2FY1rjtUVqbopbJI%3D',
      type: 'video/mp4'
    },
    options: {
      muted: true,
      controls: true
    }
  },
  {
    source: {
      src: 'https://storage.jd.com/about/big-final.mp4?Expires=3730193075&AccessKey=3LoYX1dQWa6ZXzQl&Signature=ViMFjz%2BOkBxS%2FY1rjtUVqbopbJI%3D',
      type: 'video/mp4'
    },
    options: {
      muted: true,
      controls: true
    }
  }
];

test('basic usage test', async () => {
  const wrapper = mount(ImagePreview, {
    props: {
      show: true,
      images
    }
  });
  await nextTick();
  expect((wrapper.find('.custom-pop').element as any).style.display).toEqual('');
});

test('test autoplay', async () => {
  const wrapper = mount(ImagePreview, {
    props: {
      show: true,
      images,
      autoplay: 3000
    }
  });

  await nextTick();

  expect(wrapper.vm.active).toBe(1);

  setTimeout(() => {
    expect(wrapper.vm.active).toBe(2);
  }, 3000);
});

test('init page No.', async () => {
  const wrapper = mount(ImagePreview, {
    props: {
      show: true,
      images,
      initNo: 3
    }
  });
  await nextTick();
  expect(wrapper.find('.nut-imagepreview-index').html()).toMatchSnapshot();
});

test('customize pagination and color', async () => {
  const wrapper = mount(ImagePreview, {
    props: {
      show: true,
      images,
      paginationVisible: true,
      paginationColor: 'red'
    }
  });
  await nextTick();
  const swiperPagination = wrapper.find('.nut-swiper-pagination');
  expect(swiperPagination.exists()).toBe(true);
  expect(swiperPagination.findAll('i')[0].element.style.backgroundColor).toEqual('red');
});

test('video surported in H5 env', async () => {
  const wrapper = mount(ImagePreview, {
    props: {
      show: true,
      images,
      videos
    }
  });
  await nextTick();
  expect(wrapper.find('.custom-pop').html()).toMatchSnapshot();
});
