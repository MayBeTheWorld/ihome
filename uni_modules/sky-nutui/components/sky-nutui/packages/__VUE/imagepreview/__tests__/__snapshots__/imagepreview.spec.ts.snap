// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init page No. 1`] = `"<view class=\\"nut-imagepreview-index\\">3 / 4</view>"`;

exports[`video surported in H5 env 1`] = `
"<view class=\\"nut-popup popup-center custom-pop\\" style=\\"z-index: 2005; animation-duration: 0.3s; width: 100%;\\">
  <!-- @click.stop=\\"closeOnImg\\" @touchstart.capture=\\"onTouchStart\\" -->
  <view class=\\"nut-imagepreview\\">
    <view class=\\"nut-swiper nut-imagepreview-swiper\\">
      <view class=\\"nut-swiper-inner\\">
        <view class=\\"nut-swiper-item\\">
          <!--v-if-->
          <view class=\\"nut-imagepreview-box\\">
            <div class=\\"nut-video\\"><video class=\\"nut-video-player\\" controls=\\"\\">
                <source src=\\"https://storage.jd.com/about/big-final.mp4?Expires=3730193075&amp;AccessKey=3LoYX1dQWa6ZXzQl&amp;Signature=ViMFjz%2BOkBxS%2FY1rjtUVqbopbJI%3D\\" type=\\"video/mp4\\">
              </video>
              <!--v-if-->
              <!--v-if-->
              <div class=\\"nut-video-controller show-control\\" style=\\"display: none;\\">
                <div class=\\"control-play-btn\\"></div>
                <div class=\\"current-time\\">00:00</div>
                <div class=\\"progress-container\\">
                  <div class=\\"progress\\">
                    <div class=\\"buffered\\" style=\\"width: 0%;\\"></div>
                    <div class=\\"video-ball\\" style=\\"transform: translate3d(0px, -50%, 0);\\">
                      <div class=\\"move-handle\\"></div>
                    </div>
                    <div class=\\"played\\"></div>
                  </div>
                </div>
                <div class=\\"duration-time\\">00:00</div>
                <div class=\\"volume muted\\"></div>
                <div class=\\"fullscreen-icon\\"></div>
              </div><!-- 错误弹窗 -->
              <div class=\\"nut-video-error\\" style=\\"display: none;\\">
                <p class=\\"lose\\">视频加载失败</p>
                <p class=\\"retry\\">点击重试</p>
              </div>
            </div>
          </view>
        </view>
        <view class=\\"nut-swiper-item\\">
          <!--v-if-->
          <view class=\\"nut-imagepreview-box\\">
            <div class=\\"nut-video\\"><video class=\\"nut-video-player\\" controls=\\"\\">
                <source src=\\"https://storage.jd.com/about/big-final.mp4?Expires=3730193075&amp;AccessKey=3LoYX1dQWa6ZXzQl&amp;Signature=ViMFjz%2BOkBxS%2FY1rjtUVqbopbJI%3D\\" type=\\"video/mp4\\">
              </video>
              <!--v-if-->
              <!--v-if-->
              <div class=\\"nut-video-controller show-control\\" style=\\"display: none;\\">
                <div class=\\"control-play-btn\\"></div>
                <div class=\\"current-time\\">00:00</div>
                <div class=\\"progress-container\\">
                  <div class=\\"progress\\">
                    <div class=\\"buffered\\" style=\\"width: 0%;\\"></div>
                    <div class=\\"video-ball\\" style=\\"transform: translate3d(0px, -50%, 0);\\">
                      <div class=\\"move-handle\\"></div>
                    </div>
                    <div class=\\"played\\"></div>
                  </div>
                </div>
                <div class=\\"duration-time\\">00:00</div>
                <div class=\\"volume muted\\"></div>
                <div class=\\"fullscreen-icon\\"></div>
              </div><!-- 错误弹窗 -->
              <div class=\\"nut-video-error\\" style=\\"display: none;\\">
                <p class=\\"lose\\">视频加载失败</p>
                <p class=\\"retry\\">点击重试</p>
              </div>
            </div>
          </view>
        </view>
        <view class=\\"nut-swiper-item\\">
          <view style=\\"transition-duration: .3s;\\" class=\\"nut-imagepreview-box\\"><img src=\\"//m.360buyimg.com/mobilecms/s750x366_jfs/t1/18629/34/3378/144318/5c263f64Ef0e2bff0/0d650e0aa2e852ee.jpg\\" class=\\"nut-imagepreview-img\\"></view>
          <!--v-if-->
        </view>
        <view class=\\"nut-swiper-item\\">
          <view style=\\"transition-duration: .3s;\\" class=\\"nut-imagepreview-box\\"><img src=\\"//m.360buyimg.com/mobilecms/s750x366_jfs/t1/26597/30/4870/174583/5c35c5d2Ed55eedc6/50e27870c25e7a82.png\\" class=\\"nut-imagepreview-img\\"></view>
          <!--v-if-->
        </view>
        <view class=\\"nut-swiper-item\\">
          <view style=\\"transition-duration: .3s;\\" class=\\"nut-imagepreview-box\\"><img src=\\"//m.360buyimg.com/mobilecms/s750x366_jfs/t1/9542/17/12873/201687/5c3c4362Ea9eb757d/60026b40a9d60d85.jpg\\" class=\\"nut-imagepreview-img\\"></view>
          <!--v-if-->
        </view>
        <view class=\\"nut-swiper-item\\">
          <view style=\\"transition-duration: .3s;\\" class=\\"nut-imagepreview-box\\"><img src=\\"//m.360buyimg.com/mobilecms/s750x366_jfs/t1/30042/36/427/82951/5c3bfdabE3faf2f66/9adca782661c988c.jpg\\" class=\\"nut-imagepreview-img\\"></view>
          <!--v-if-->
        </view>
      </view>
      <!--v-if-->
    </view>
  </view>
  <view class=\\"nut-imagepreview-index\\">1 / 6</view>
  <!--v-if-->
  <!--v-if-->
</view>"
`;
